tests/optimizers_test.py

```diff
@@ -138,7 +138,14 @@ def loss(xs):
     x0 = (jnp.ones(2), jnp.ones((2, 2)))
     self._CheckOptimizer(optimizers.adagrad, loss, x0, num_iters, step_size)
 
-  def testSM3(self):
+  def testSM3Scalar(self):
+    def loss(x): return x**2
+    x0 = 1.
```

Comment:
I think if you use `x0 = jnp.array(1.)` it will avoid the error in the tests.

Reply:
Updated